Process for maintaining Build + CI contributors list (#83869)
pytorch/pytorch: docs/source/community/build_ci_governance.rst

PyTorch Governance | Build + CI
===============================

> How to Add New Maintainer

Reviewer: ```suggestion
How to Add a New Maintainer
```

Author: fixed

> -------------------------

Reviewer: ```suggestion
---------------------------
```

> For the person to be a maintainer, a person needs to:

Reviewer: ```suggestion
For the person to be a maintainer, a person needs to:

```

Author: fixed

> * land at least six commits to the related part of the PyTorch

Reviewer: ```suggestion
* land at least six commits to the related part of the PyTorch repository
```

> * at least one of these commits should be in the last six months

Reviewer: ```suggestion
* at least one of these commits must be submitted in the last six months
```

Author: fixed

To add a qualified person to the maintainers' list, please create

> a PR that adds a person to the "people of interests" page and

Reviewer: ```suggestion
a PR that adds a person to the `persons of interests <https://pytorch.org/docs/master/community/persons_of_interest.html>`__ and 
```

Author: done

> merge_rules files. Current maintainers will cast their votes of

Reviewer: ```suggestion
`merge_rules <https://github.com/pytorch/pytorch/blob/master/.github/merge_rules.json>`__ files. Current maintainers will cast their votes of
```

Author: done

support. Decision criteria for approving the PR:

> * no earlier than two business days passed before merging (make

Reviewer: ```suggestion
* Not earlier than two business days passed before merging (make
```

Author: done

  sure that the majority of the contributors have seen it)

> * PR has the correct label ("module: ci")

Reviewer: ```suggestion
* PR has the correct label (`module: ci`)
```

Author: done

> * there are no objections from the current maintainers

Reviewer: ```suggestion
* There are no objections from the current maintainers
```

Author: done

> * there are at least three "thumbs up" from current maintainers

Reviewer: ```suggestion
* There are at least three *thumbs up* from current maintainers
```

Author: done

PyTorch Governance | Build + CI
===============================

How to Add a New Maintainer
---------------------------

> For the person to be a maintainer, a person needs to:

Reviewer 2: Can we include a definition of what we think a maintainer is and what their responsibilities/privileges are?  The difference between a "maintainer" and a "contributor" seems important here.

Clearly, simply making any six commits isn't enough for a person to qualify. That's why we have a voting mechanism included

> * land at least six commits to the related part of the PyTorch repository

Reviewer 2: This wording can be a bit more generalized.  Our main gate keeper is the votes

```suggestion
* land at least six commits or significant contributions to the related part of the PyTorch repository
```

> * land at least six commits to the related part of the PyTorch repository
> * at least one of these commits must be submitted in the last six months

Reviewer 2: Nits

```suggestion
* Land at least six commits to the related part of the PyTorch repository
* At least one of these commits must be submitted in the last six months
```

To add a qualified person to the maintainers' list, please create
a PR that adds a person to the `persons of interests <https://pytorch.org/docs/master/community/persons_of_interest.html>`__ page and
`merge_rules <https://github.com/pytorch/pytorch/blob/master/.github/merge_rules.json>`__ files. Current maintainers will cast their votes of
support. Decision criteria for approving the PR:
* Not earlier than two business days passed before merging (make
  sure that the majority of the contributors have seen it)
* PR has the correct label (`module: ci`)
* There are no objections from the current maintainers

> * There are at least three *thumbs up* from current maintainers

Reviewer 2: Might be useful to add a criteria for a "maintainer" to be demoted to "emeritus".

E.g. "If at least six months go by without a maintainer making contributions to pytorch, then their status will be changed to the honorary emeritus"

> * There are at least three *thumbs up* from current maintainers

Reviewer 2: What if there are *thumbs down* votes? Or if there are less than three maintainers?

```suggestion
* There are at least three net *thumbs up* from current maintainers (or all maintainers vote *thumbs up* when the module has less than 3 maintainers)
```
